src/components/forms/supplierMaster/SupplierMasterForm.tsx

```diff
@@ -0,0 +1,177 @@
+import React from 'react';
+import { useForm } from 'react-hook-form';
+import { RiCloseFill } from 'react-icons/ri';
+import { IconContext } from 'react-icons';
+import Input from '../../../shared/inputs/input';
+import SubmitButton from '../../../shared/buttons/SubmitButton';
+import CancelButton from '../../../shared/buttons/CancelButton';
+import ValidationError from '../../shared/ValidationError';
+import DeleteButton from '../../../shared/buttons/DeleteButton';
+import SupplierFormType from './types';
+
+function SupplierMasterForm({
+  onSubmit, onCancel, title, submitButtonText, initialFormValue, mode,
+}: SupplierFormType) {
```

Comment:
Can you use ManualFormType ? (you can find in form/types.tsx)

Reply:
Every form cannot have the same type, because of different types in redux